openssl-dynamic/src/main/c/jnilib.c

```diff
@@ -291,27 +291,41 @@ static const jint method_table_size = sizeof(method_table) / sizeof(method_table
 // JNI Method Registration Table End
 
 static jint netty_internal_tcnative_Library_JNI_OnLoad(JNIEnv* env, const char* packagePrefix) {
+    int errorOnLoadCalled = 0;
```

Comment:
for coolness points, you can make these labels and goto the correct one:

```
bufferOnLoadCalled: netty_internal_tcnative_Buffer_JNI_OnUnLoad(env);
errorOnLoadCalled: netty_internal_tcnative_Error_JNI_OnUnLoad(env);

```

Reply:
You "fall through" from one label to the next, so it will call multiple unload methods naturally.

Reply:
Ah got it... not sure this is clearer tho. I think I would prefer to keep it as it is